app/controllers/report_controller/schedules.rb

```diff
@@ -241,7 +241,7 @@ def schedule_edit
       add_flash(_("All changes have been reset"), :warning) if params[:button] == "reset"
       if x_active_tree != :reports_tree
         # dont set these if new schedule is being added from a report show screen
-        obj = find_checked_ids_with_rbac(MiqSchedule)
+        obj = find_records_with_rbac(MiqSchedule, checked_or_params)
         obj[0] = params[:id] if obj.blank? && params[:id]
         @schedule = obj[0] && params[:id] != "new" ? MiqSchedule.find(obj[0]) :
             MiqSchedule.new(:userid => session[:userid])  # Get existing or new record
```

Comment:
```ruby
@schedule = obj[0] && params[:id] != "new" ? MiqSchedule.find(obj[0]) : MiqSchedule.new(:userid => session[:userid])  # Get existing or new record
```

This line tells us, that we either create a new `MiqSchedule` or accessing the database (again) to load the record (with the id that should be in `obj`).